kyo-core/shared/src/main/scala/kyo/Channel.scala

```diff
@@ -158,6 +158,54 @@ object Channel:
           */
         def full(using Frame): Boolean < (Abort[Closed] & IO) = IO.Unsafe(Abort.get(self.full()))
 
+        private def emitChunks(maxChunkSize: Maybe[Int] = Maybe.Absent)(using
+            Tag[A],
```

Comment:
The `Tag` implementation is able to avoid allocations but only when the tag is fully known statically. In cases where it needs to be derived, an allocation is introduced. This should be a concern in user code but it's an important optimization for Kyo's primitives.
```suggestion
            Tag[Emit[Chunk[A]]],
```